libs/output-mapping/src/Writer/Table/TableConfigurationValidator.php

```diff
@@ -7,41 +7,95 @@
 use InvalidArgumentException;
 use Keboola\OutputMapping\Exception\InvalidOutputException;
 use Keboola\OutputMapping\Mapping\MappingFromRawConfigurationAndPhysicalDataWithManifest;
+use Keboola\OutputMapping\OutputMappingSettings;
 use Keboola\OutputMapping\Writer\Helper\RestrictedColumnsHelper;
 use Keboola\OutputMapping\Writer\Table\Strategy\SqlWorkspaceTableStrategy;
 
 class TableConfigurationValidator
 {
+    public function __construct(
+        readonly private StrategyInterface $strategy,
+        readonly private OutputMappingSettings $settings,
+    ) {
+    }
+
     public function validate(
-        StrategyInterface $strategy,
         MappingFromRawConfigurationAndPhysicalDataWithManifest $source,
         array $config,
     ): void {
-        if (!$strategy instanceof SqlWorkspaceTableStrategy) {
-            try {
-                RestrictedColumnsHelper::validateRestrictedColumnsInConfig(
-                    $config['columns'],
-                    $config['column_metadata'],
-                );
-            } catch (InvalidOutputException $e) {
+        if (!$this->strategy instanceof SqlWorkspaceTableStrategy) {
+            $this->validateRestrictedColumns($source, $config);
+        }
+
+        if ($this->settings->hasNewNativeTypesFeature()) {
+            $this->validateSchemaConfig($config);
+        }
+
+        $this->validateColumnsSpecification($source, $config);
+
+        $this->validateMappingDestination($config);
+    }
+
+    private function validateRestrictedColumns(
+        MappingFromRawConfigurationAndPhysicalDataWithManifest $source,
+        array $config,
+    ): void {
+        try {
+            RestrictedColumnsHelper::validateRestrictedColumnsInConfig(
+                $config['columns'],
+                $config['column_metadata'],
+                $config['schema'],
+            );
+        } catch (InvalidOutputException $e) {
+            throw new InvalidOutputException(
+                sprintf(
+                    'Failed to process mapping for table %s: %s',
+                    $source->getSourceName(),
+                    $e->getMessage(),
+                ),
+                0,
+                $e,
+            );
+        }
+    }
+
+    private function validateSchemaConfig(array $config): void
+    {
+        if ($this->settings->getDataTypeSupport() === OutputMappingSettings::DATA_TYPES_SUPPORT_NONE) {
+            return;
+        }
+
+        if (empty($config['schema'])) {
+            throw new InvalidOutputException('Configuration schema is missing.');
+        }
+
+        if ($this->settings->getDataTypeSupport() === OutputMappingSettings::DATA_TYPES_SUPPORT_AUTHORITATIVE) {
+            $missingDataType = array_filter($config['schema'], fn($column) => !isset($column['data_type']['base']));
+
```

Comment:
tady by možná měla být ještě jedna věc - že když je tam definovaný backend, tak pro ten backend by taky měly být specifickovaný všechny sloupce (jakože z 10 sloupců v tabulce nesmí být jen pro dva specifikovanej snowflake data type, ale musí být pro všechny)

Reply:
Jakože abysme použili buď basetype nebo backend ale pro všechny sloupce a ne pro každej individuálně - nebo myslíš, že by to bylo moc tvrdý?

Reply:
tvrdý to není, spíš si myslím že takhle to je validní... když máš 10 sloupců a chceš jen pro jeden z nich u snflk dát DATETIME, tak je blbost pak pro všechny sloupečky definovat typ pro snflk když u zbytku by mohl být stejný jako base_type

Reply:
jo, po uživateli nebudem chít, aby si to přetěžoval všechno